powerauth-push-server/src/main/java/io/getlime/push/service/PushMessageSenderService.java

```diff
@@ -278,6 +293,29 @@ private void updateStatusAndPersist(PushMessageEntity pushMessageObject, PushMes
         }
     }
 
+    private ApnsEnvironment resolveApnsEnvironment(final String environmentDevice, final String environmentAppConfig) {
+        if (environmentDevice != null) {
+            final ApnsEnvironment envDevice = ApnsEnvironment.fromString(environmentDevice);
+            if (!configuration.isApnsUseDevelopment() && envDevice == ApnsEnvironment.DEVELOPMENT) {
+                logger.warn("Invalid configuration: server is configured with APNs in production mode, however push device is registered in development mode.");
```

Comment:
The logging is excessive either log error after the call of the resolveEnvironment, or log warn in the resolveEnvironment.

Reply:
Ok, I will move the code outside of this method.